src/loki/trace_route_action.cc

```diff
@@ -40,7 +40,18 @@ void check_shape(const google::protobuf::RepeatedPtrField<valhalla::Location>& s
 
 void check_distance(const google::protobuf::RepeatedPtrField<valhalla::Location>& shape,
                     float max_distance,
+                    float max_breakage_distance,
                     float max_factor = 1.0f) {
+
+  for (auto iter = shape.cbegin(); iter < shape.cend() - 1; ++iter) {
+    // We bail when the distance between two locations exceeds the threshold
+    PointLL curr_point = to_ll(*iter);
+    PointLL next_point = to_ll(*std::next(iter));
+    if (curr_point.Distance(next_point) > max_breakage_distance) {
+      throw valhalla_exception_t{154};
```

Comment:
i dont thnk we should throw on just one. what we want to do is throw if all o fthem were above the breakage distance because then we know we can never even do a match.

additionally we want to keep track of the total distance and throw if we ever excede that rather than checking the first and last points only